glusterd2/commands/volumes/volume-reset.go

```diff
@@ -0,0 +1,111 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerVolResetStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"vol-reset.Store", storeVolume},
+		{"vol-reset.NotifyVolfileChange", notifyVolfileChange},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volname := mux.Vars(r)["volname"]
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	var req api.VolOptionReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	for _, k := range req.Options {
+		if _, ok := volinfo.Options[k]; ok {
+			err = txn.Ctx.Delete(k)
```

Comment:
Before deleting from volinfo, check is required for `VOLOPT_FLAG_FORCE` and `VOLOPT_FLAG_NEVER_RESET` https://github.com/gluster/glusterd2/pull/545#issuecomment-362215319

- Error if VOLOPT_FLAG_NEVER_RESET is set and option is trying to reset.
- Only allow Reset if force is used in API request in case of VOLOPT_FLAG_FORCE

Reply:
@aravindavk am not quite sure how to get these volopt flags' values. I intend to ask if those flags happen to be part of the volinfo or ..?
Checking glusterfs code, I found them listed as follows:
```
        { .key         = VKEY_FEATURES_QUOTA,
          .voltype     = "features/marker",
          .option      = "quota",
          .value       = "off",
          .type        = NO_DOC,
          .flags       = VOLOPT_FLAG_NEVER_RESET,
          .op_version  = 1
        },

```